code/game/objects/items/robot/items/generic.dm

```diff
@@ -102,41 +121,60 @@
+				SEND_SIGNAL(attacked_mob, COMSIG_BORG_TOUCH_MOB)
 				playsound(loc, 'sound/weapons/tap.ogg', 50, TRUE, -1)
 			else if(ishuman(attacked_mob))
 				if(user.body_position == LYING_DOWN)
-					user.visible_message(span_notice("[user] shakes [attacked_mob] trying to get [attacked_mob.p_them()] up!"), \
-									span_notice("You shake [attacked_mob] trying to get [attacked_mob.p_them()] up!"))
+					user.visible_message(
+						span_notice("[user] shakes [attacked_mob] trying to get [attacked_mob.p_them()] up!"), 
+						span_notice("You shake [attacked_mob] trying to get [attacked_mob.p_them()] up!"),
+					)
 				else
-					user.visible_message(span_notice("[user] hugs [attacked_mob] to make [attacked_mob.p_them()] feel better!"), \
-							span_notice("You hug [attacked_mob] to make [attacked_mob.p_them()] feel better!"))
+					user.visible_message(
+						span_notice("[user] hugs [attacked_mob] to make [attacked_mob.p_them()] feel better!"),
+						span_notice("You hug [attacked_mob] to make [attacked_mob.p_them()] feel better!"),
+					)
+					SEND_SIGNAL(attacked_mob, COMSIG_BORG_TOUCH_MOB)
 				if(attacked_mob.resting)
 					attacked_mob.set_resting(FALSE, TRUE)
 			else
-				user.visible_message(span_notice("[user] pets [attacked_mob]!"), \
-						span_notice("You pet [attacked_mob]!"))
+				user.visible_message(
+					span_notice("[user] pets [attacked_mob]!"),
+					span_notice("You pet [attacked_mob]!"),
+				)
 			playsound(loc, 'sound/weapons/thudswoosh.ogg', 50, TRUE, -1)
 		if(HUG_MODE_HUG)
 			if(ishuman(attacked_mob))
 				attacked_mob.adjust_status_effects_on_shake_up()
 				if(attacked_mob.body_position == LYING_DOWN)
-					user.visible_message(span_notice("[user] shakes [attacked_mob] trying to get [attacked_mob.p_them()] up!"), \
-									span_notice("You shake [attacked_mob] trying to get [attacked_mob.p_them()] up!"))
+					user.visible_message(
+						span_notice("[user] shakes [attacked_mob] trying to get [attacked_mob.p_them()] up!"),
+						span_notice("You shake [attacked_mob] trying to get [attacked_mob.p_them()] up!"),
+					)
 				else if(user.zone_selected == BODY_ZONE_HEAD)
-					user.visible_message(span_warning("[user] bops [attacked_mob] on the head!"), \
-									span_warning("You bop [attacked_mob] on the head!"))
+					user.visible_message(span_warning("[user] bops [attacked_mob] on the head!"),
+						span_warning("You bop [attacked_mob] on the head!"),
+					)
+					SEND_SIGNAL(attacked_mob, COMSIG_BORG_TOUCH_MOB)
 					user.do_attack_animation(attacked_mob, ATTACK_EFFECT_PUNCH)
 				else
-					user.visible_message(span_warning("[user] hugs [attacked_mob] in a firm bear-hug! [attacked_mob] looks uncomfortable..."), \
-							span_warning("You hug [attacked_mob] firmly to make [attacked_mob.p_them()] feel better! [attacked_mob] looks uncomfortable..."))
+					if(~SEND_SIGNAL(attacked_mob, COMSIG_BORG_HUG_MOB, user) & COMSIG_BORG_HUG_SUCCESS)
```

Comment:
"Borg hug success" means even less to me in this context. I'm not sure what you're doing here.

Also don't use `~` for this. If you want to invert the check, then `if (!(send_signal() & HUG_SUCCESS))`